### DISCUSSION TITLE
MODE-TD percentiles

### DISCUSSION THREAD

**Discussion by skmue** [dtcenter/METplus#1541] [Statistical Computation] [answered]
Dear MET help team,

I am writing to in concerns of the computation of the intensity percentiles in MODE-TD. When applying a threshold to precipitation rate of 5mm/hour (see attached files and plot) I yield the following intensity percentiles for object F026: [p10,p25,p50,p75,p90,p99] = [0.01,0.41,3.77,15.57,33.43,79.34]

I was surprised that the small percentiles are <5mm/hour, as I had assumed that MODE-TD computes the percentiles from the object's convolved field, which would be all greater 5mm/hour. I now see that MODE-TD computes the percentiles from the raw field, but I find this unfortunate, since then it is integrated over grid cells that actually don't show precipitation at all. Eventually I'd like to compute the mean precipitation rate or integrated rain volume of an object, and I, just like MODE-TD, do this by integrating the original/raw field over the objects identified cells, which I fear of being in-accurate.

Well, I would just like to raise some concerns about this. I guess to compute percentiles from the convolved field would be the better solution.

Best regards
Sebastian
[mtd_ALP-3i_ICTP_RegCM4_rcp85_prTH05_rsmoo8_tsmoo1_minvol100_2098090100__20980901_003000V_3d_single_simple.txt](https://github.com/dtcenter/METplus/files/8390628/mtd_ALP-3i_ICTP_RegCM4_rcp85_prTH05_rsmoo8_tsmoo1_minvol100_2098090100__20980901_003000V_3d_single_simple.txt)
[mtd_ALP-3i_ICTP_RegCM4_rcp85_prTH05_rsmoo8_tsmoo1_minvol100_2098090100__20980901_003000V_2d.txt](https://github.com/dtcenter/METplus/files/8390629/mtd_ALP-3i_ICTP_RegCM4_rcp85_prTH05_rsmoo8_tsmoo1_minvol100_2098090100__20980901_003000V_2d.txt)
![Concept_Tracker](https://user-images.githubusercontent.com/94064230/161086512-ff7196ef-acac-4bcb-bd20-60b5417b88b0.png)

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your inquiry. As you have already determined, the default for MODE-TD is to calculate the intensity percentiles on the raw data. We do not intend to change this behavior, but after conferring with one of our lead engineers, I can provide the following guidance:

> However, I do think there's a way the user could reconfigure MTD to achieve the desired result with the following...
Use the regrid dictionary to replace the raw values with the same thing the convolution step would produce:

```
regrid = {
   to_grid    = FCST;
   method     = UW_MEAN;
   width      = 5; // or whatever
   vld_thresh = 0.5;
   shape     = CIRCLE;
}
```

> Disable the convolution smoothing step by setting the radius to 1.

`conv_radius = 0.5; // or 0?`

> But I'd recommend testing this. In MODE the logic is a little simpler. Setting the radius to 0 tells it to skip the smoothing step. In MTD, which smooths in time and space, you probably still want to smooth in time... but just using the (already smoothed) single grid point for each time.

I hope this can provide some guidance to make MODE-TD do something closer to what you desire.

**Comment by j-opatz** [comment]
Hello @skmue,

I wanted to check in with you about the initial discussion you submitted. We were able to provide some guidance for you on April 1, but haven't heard back if the suggested actions helped. 
If it did, or if you found a different way of accomplishing your goal, we'd appreciate hearing from you. If you have additional questions on this, we can try to find a solution; otherwise we'll close this Discussion due to inactivity.

**Comment by skmue** [comment]
Hello @j-opatz,

two questions remain here: 
1) does the proposed method compute statistics from the smoothed field then?
   - I guess the answer to this is, no.
2) What is the complete regrid command to the MODE-TD convolution?
>    conv_time_window  = { beg = -1; end = 1; };
>    conv_radius       = 8;

Eventually, if I could only do the regrid without using the MODE-TD, that would help.
Else, I am happy to know what MODE-TD is doing now and I don't believe the analysis is much affected anyway, or at least it is systematically affected.

Many thanks and cheers!
Sebastian

**Reply by j-opatz** [reply]
For the two remaining questions,

1. No is correct. The behavior of MODE-TD is done intentionally and development doesn't see an overwhelming demand to change that behavior at this time.
2. This part can be tricky, as it depends on what you want your end result to be. The `regrid` dictionary effects are applied to the raw fields **before** the intensity percentiles are calculated, so using the settings I described should modify the raw fields to mimic how the convolved fields look. However, there's the assumption that you want to smooth at least over the temporal domain. Hence, you don't want to set the `conv_radius` to 0 as that would disable all smoothing. Keeping `conv_radius` at 1 or less as suggested in my previous post keeps the smoothing on and allows you access to the `conv_time_window`  option as you've listed, but it eliminates the spatial smoothing. What `conv_time_window`needs to be set to is ultimately up to you, though. Do you expect a temporal smoothing for the convolved fields of the previous and next time steps only (i.e. `conv_time_window = { beg = -1; end = 1; };`)? Or do you need a wider window? Those are controlled with `conv_time_window`.
As for your question,

> Else, I am happy to know what MODE-TD is doing now

This is 100% determined by what's used in the MODE-TD configuration file. A quick look in the MET [Github repository](https://github.com/dtcenter/MET/blob/main_v10.1/met/data/config/MTDConfig_default) shows the default values for the MODE-TD are:

```
regrid = {
   to_grid    = NONE;
   method     = NEAREST;
   width      = 1;
   vld_thresh = 0.5;
   shape     = SQUARE;
}
```

and

```
censor_thresh     = [];
   censor_val        = [];
   conv_time_window  = { beg = -1; end = 1; };
   conv_radius       = 60.0/grid_res; // in grid squares
   conv_thresh       = >=5.0;

```

so the default is no regridding (the raw fields are unchanged) and the convolved field is temporally smoothed from the previous and next time steps, and spatially smoothed by 60.0/grid_res (grid_res is set to 4), with anything greater than or equal to 5.0 being defined as an object.
Hopefully that helps you understand a little more about the process and the remaining settings in the configuration file. If not, please let me know and I can reach out to our engineers to better assist you.

**Reply by j-opatz** [reply]
@skmue Were there any further questions on this topic? Otherwise we'd like to close the Discussions and prevent it from going stale.
